src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -631,6 +631,11 @@ public void addToCallbackEvents(MqttClientConnectionEvents callbacks) {
         callbackEventManager.addToCallbackEvents(callbacks);
     }
 
+    public void addToCallbackEvents(CallbackEventManager.OnConnectCallback onConnect,
+                                    MqttClientConnectionEvents callbacks) {
+        callbackEventManager.addToCallbackEvents(onConnect, callbacks);
+    }
```

Comment:
Why do you need this to be exposed? You should simply assume that the client is always connected.